### ISSUE TITLE
[BUG] Phalcon\Mvc\Model\Criteria::inWhere doesn't accept empty array

### ISSUE THREAD

**Issue by b-viguier** [phalcon/cphalcon#1292]
In Phalcon\Mvc\Model\Criteria::inWhere, if the second parameter is an empty array, the generated PHQL query is not valid.

**Behaviour**: an exception is thrown:
```
Phalcon\Mvc\Model\Exception' with message 'Syntax error, unexpected token ), near to ')', when parsing: SELECT [XXX].* FROM [XXX] WHERE (_id IN ()) (120)' in XXX
```

**Expected Behaviour**: the query should be valid, and return an empty resultset once executed.

Code location: https://github.com/phalcon/cphalcon/blob/master/ext/mvc/model/query/builder.c#L956

*PS: It's my 1st post: awesome work for this framework :)*

**Comment by stgatev** [comment]
I have the same problem, and I like the proposed solution. In my view, inWhere() should NOT throw an error when the array is empty, and return an empty resultset instead. Since we are abstracting the actual SQL, the framework should take care of the corner cases.

**Comment by andresgutierrez** [comment]
Fixed in Phalcon 2.0.4
